app/services/cold_storage_backends/s3_backend.py

```
@@ -0,0 +1,157 @@
+            client = self._client(location)
+
+            # Calculate checksum BEFORE upload/deletion so it is never None for MOVE.
+            checksum_before = checksum_verifier.calculate_checksum(source_path)
+
+            client.upload_file(str(source_path), bucket, key)
+
+            if operation_mode == OperationType.MOVE:
+                source_path.unlink()
+
+            return True, None, f"s3://{bucket}/{quote(key)}", checksum_before
+        except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)